Data/track order event properties for extensions

jaclync opened the pull request:
<!-- Remember about a good descriptive title. -->

Part of #10337 
<!-- Id number of the GitHub issue this PR addresses. -->

## Why
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

To better prioritize the extensions we want to support in the app, it'd be great to understand how often they are used in the orders/products viewed by the merchants. The top priority extensions sorted by active stores count in the last 30 days that also use mobile:

- WooCommerce Product Add-ons: how many opened orders have line items with product add-ons - `order_products_loaded.has_addons` (new event & property) when the order's products are synced
- WooCommerce Product Bundles: how many opened orders have line items with a bundle product - `order_products_loaded.product_types: comma-delimited list string` (new event & property) when the order's products are synced
- WooCommerce Subscriptions: how many opened orders have line items with a subscription product - same as above
- WooCommerce Gift Cards: how many opened orders include a gift card redemption - ✅  `order_details_gift_card_shown` when gift card is shown in order details
- WooCommerce Min/Max Quantities: how many opened products include min/max quantity rules - `product_detail_loaded.has_minmax_quantity_rules` (existing event, new property) when a product is loaded

While testing the `order_open` event, I noticed that it wasn't tracked when tapping on an order from the order search screen. Now it should be tracked in the PR.

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

### `order_products_loaded` event (new)

- Go to the orders tab
- Tap on an order that includes at least one subscription product --> an event `order_products_loaded` should be tracked with the subscription product type in the `product_types` property (e.g. `variable-subscription`)
- Tap on an order that includes at least one bundle product --> an event `order_products_loaded` should be tracked with the `bundle` product type in the `product_types` property
- Tap on an order that includes at least one product with add-ons --> an event `order_products_loaded` should be tracked with `has_addons: true`
- Tap on an order that doesn't include any product with add-ons --> an event `order_products_loaded` should be tracked with `has_addons: false`

### `product_detail_loaded` event

- Go to the products tab
- Tap on a product with min/max quantity rules --> an event `product_detail_loaded` should be tracked with `has_minmax_quantity_rules: true`
- Tap on a product without min/max quantity rules --> an event `product_detail_loaded` should be tracked with `has_minmax_quantity_rules: false`

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

jaclync committed:
Track an event property on whether an order includes a gift card redemption.

jaclync committed:
Track `order_products_loaded` to analyze the product types in order details.

jaclync committed:
Track `order_products_loaded.has_addons` for analyzing product add-ons usage in order details.

jaclync committed:
Track `product_detail_loaded.has_minmax_quantity_rules` to analyze Min/Max Quantities extension usage in order details.

jaclync committed:
Track `order_open` event also when tapping an order from the order search results.

rachelmcr reviewed (approved):
Works great! I'm approving because it all looks good and works as described, but I have a couple questions just to check before this is merged:

Have we already registered the new event `order_products_loaded` and updated the other events to describe their new properties?

> WooCommerce Gift Cards: how many opened orders include a gift card redemption - order_open.redeemed_gift_card (existing event, new property) when an order is loaded

As I mentioned in Slack, we do have `order_details_gift_card_shown` which is tracked when gift cards are shown in order details. I can see how the new property on `order_open` is useful to explicitly track if an order that is opened has redeemed gift cards in it, but we could also use `order_details_gift_card_shown`  compared to `order_open` events to calculate that now (without having to wait for a new event). Given that, what do you think — is the new property needed?

> While testing the order_open event, I noticed that it wasn't tracked when tapping on an order from the order search screen. Now it should be tracked in the PR.

Nice find!

jaclync committed:
Remove `order_open.redeemed_gift_card` property since we already track an event for the same data.

jaclync commented:
> As I mentioned in Slack, we do have `order_details_gift_card_shown` which is tracked when gift cards are shown in order details. I can see how the new property on `order_open` is useful to explicitly track if an order that is opened has redeemed gift cards in it, but we could also use `order_details_gift_card_shown` compared to `order_open` events to calculate that now (without having to wait for a new event). Given that, what do you think — is the new property needed?

Thanks for sharing the existing event when the gift card is shown in order details! We don't need the new event property for `order_open` then, I removed it in https://github.com/woocommerce/woocommerce-ios/pull/10332/commits/4268ca6e47e22897442561dba2a3e0760a8e5cb5 and updated the PR description.

> Have we already registered the new event order_products_loaded and updated the other events to describe their new properties?

Nice that Eagle still has the process to register events, I don't think we do that in Kiwi but we probably should 😅 Just registered `product_detail_loaded.has_minmax_quantity_rules` in https://github.com/Automattic/tracks-events-registration/pull/1745, feel free to take a look! Other default event properties were also added automatically, lemme know if we prefer not to have those common properties on the registration page.  
I still can't see the new event `order_products_loaded` in the Tracks Live view or the events directory, only when searching in the history. Maybe it takes another day or two to appear. I'll make a reminder to register for the new event.

Leaving the PR open until later my day if you'd like another look!

jaclync commented:
Merging this now for code freeze, but feel free to leave any follow-up comments!

rachelmcr commented:
Looks good, thanks for following up on the event registration! I see `order_products_loaded` in the events directory now so it can be registered whenever you're ready. 👍

jaclync commented:
Just [registered the `order_products_loaded` event](https://github.com/Automattic/tracks-events-registration/pull/1774)!
